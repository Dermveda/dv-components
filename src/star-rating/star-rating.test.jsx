import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import StarRating from './star-rating.jsx';
const question = {
	_id: 'abc123',
	name: 'test q',
	type: 'test',
	isCorrect: true,
	answers: [
		{
			_id: 'abc234',
			name: 'test a',
			isChosen: true,
			isCorrect: false
		},
		{
			_id: 'abc234',
			name: 'test a',
			isChosen: true,
			isCorrect: false
		},
		{
			_id: 'abc234',
			name: 'test a',
			isChosen: true,
			isCorrect: false
		},
		{
			_id: 'abc234',
			name: 'test a',
			isChosen: true,
			isCorrect: false
		},
		{
			_id: 'abc234',
			name: 'test a',
			isChosen: true,
			isCorrect: false
		}
	]
};
describe('<StarRating/>', () => {
	it('Will render', () => {
		const el = renderer.create(<StarRating question={question} />);
		expect(el).toMatchSnapshot();
	});
	it('Will have question', () => {
		const el = shallow(<StarRating question={question} />);
		expect(el).toMatchSnapshot();
	});
	it('Will select answer', () => {
		const func = jest.fn();
		const el = mount(<StarRating question={question} onSelectAnswer={func} />);
		el
			.find('i.material-icons')
			.first()
			.simulate('click');
		expect(func).toHaveBeenCalledTimes(1);
	});
});
