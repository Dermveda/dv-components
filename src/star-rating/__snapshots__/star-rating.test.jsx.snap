// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StarRating/> Will have question 1`] = `
<li
  key="qq_star_abc123"
>
  test q
  <div
    className="dvc-pointer"
  >
    <i
      className="material-icons"
      key="star_rating_0"
      onClick={[Function]}
      style={
        Object {
          "fontSize": "200%",
        }
      }
    >
      star_border
    </i>
    <i
      className="material-icons"
      key="star_rating_1"
      onClick={[Function]}
      style={
        Object {
          "fontSize": "200%",
        }
      }
    >
      star_border
    </i>
    <i
      className="material-icons"
      key="star_rating_2"
      onClick={[Function]}
      style={
        Object {
          "fontSize": "200%",
        }
      }
    >
      star_border
    </i>
    <i
      className="material-icons"
      key="star_rating_3"
      onClick={[Function]}
      style={
        Object {
          "fontSize": "200%",
        }
      }
    >
      star_border
    </i>
    <i
      className="material-icons"
      key="star_rating_4"
      onClick={[Function]}
      style={
        Object {
          "fontSize": "200%",
        }
      }
    >
      star_border
    </i>
  </div>
</li>
`;

exports[`<StarRating/> Will render 1`] = `
<li>
  test q
  <div
    className="dvc-pointer"
  >
    <i
      className="material-icons"
      onClick={[Function]}
      style={
        Object {
          "fontSize": "200%",
        }
      }
    >
      star_border
    </i>
    <i
      className="material-icons"
      onClick={[Function]}
      style={
        Object {
          "fontSize": "200%",
        }
      }
    >
      star_border
    </i>
    <i
      className="material-icons"
      onClick={[Function]}
      style={
        Object {
          "fontSize": "200%",
        }
      }
    >
      star_border
    </i>
    <i
      className="material-icons"
      onClick={[Function]}
      style={
        Object {
          "fontSize": "200%",
        }
      }
    >
      star_border
    </i>
    <i
      className="material-icons"
      onClick={[Function]}
      style={
        Object {
          "fontSize": "200%",
        }
      }
    >
      star_border
    </i>
  </div>
</li>
`;
